### PR TITLE
Feature/explore entries

### PR DESCRIPTION
**Pull request by thenav56**
Addresses
- [x] https://github.com/the-deep/server/issues/1242
- [x] https://github.com/the-deep/server/issues/1241
- [x] https://github.com/the-deep/server/issues/1240
- [x] https://github.com/the-deep/server/issues/1262
- [x] Use explore filter in generic projects export
- [x] Add excludeProjectsLessThan filter in projects and publicProjects node

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/deep_explore/tasks.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        INSERT INTO "{_tb(EntriesCountByGeoAreaAggregate)}" (
+            project_id,
+            geo_area_id,
+            date,
+            entries_count
+        )
+        (
+            SELECT
+              "{_tb(Entry)}".project_id AS project_id,
+              jsonb_array_elements_text(
```
Comment (k9845) [inline apps/deep_explore/tasks.py]:
```suggestion
             JSONB_ARRAY_ELEMENTS_TEXT (
```

File: apps/deep_explore/tasks.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            cursor.execute(get_update_entries_count_by_geo_area_aggregate_sql(), params)
+        logger.info(f"Successfull. Runtime: {time.time() - start_time} seconds")
+        tracker.value = until_date
+        logger.info(f"Saving date {tracker.value} as last tracker")
+        tracker.save()
+
+
+@shared_task
+@redis_lock('update_deep_explore_entries_count_by_geo_aggreagate')
+def update_deep_explore_entries_count_by_geo_aggreagate_task():
```
Comment (k9845) [inline apps/deep_explore/tasks.py:265]: Add to cronjob

File: apps/project/filter_set.py [hunk 3]
```diff
@@ -260,22 +257,20 @@ def filter_title(self, qs, _, value):
...
-            entry_count=models.functions.Coalesce(models.Subquery(
-                Entry.objects.filter(
-                    project=models.OuterRef('id')
-                ).order_by().values('project').annotate(
-                    count=models.Count('id', distinct=True)
-                ).values('count')[:1],
-                output_field=models.IntegerField()
-            ), 0)
-        )
+    def filter_exclude_entry_less_than(self, qs, _, value):
```
Comment (k9845) [inline apps/project/filter_set.py]: `def filter_is_test(self, qs, _, value):`
Use `qs` instead of `Project.objects`

File: deep/caches.py [hunk 4]
```diff
@@ -16,3 +21,64 @@ class CacheKey:
...
+        TOTAL_LEADS = _PREFIX + 'TOTAL-LEADS'
+        TOTAL_ENTRIES = _PREFIX + 'TOTAL-ENTRIES'
+        TOTAL_ACTIVE_USERS = _PREFIX + 'TOTAL-ACTIVE-USERS'
+        TOTAL_AUTHORS = _PREFIX + 'TOTAL-AUTHORS'
+        TOTAL_PUBLISHERS = _PREFIX + 'TOTAL-PUBLISHERS'
+        TOP_TEN_AUTHORS = _PREFIX + 'TOP-TEN-AUTHORS'
+        TOP_TEN_PUBLISHERS = _PREFIX + 'TOP-TEN-PUBLISHERS'
+        TOP_TEN_FRAMEWORKS = _PREFIX + 'TOP-TEN-FRAMEWORKS'
+        TOP_TEN_PROJECTS_BY_USERS = _PREFIX + 'TOP-TEN-PROJECTS-BY-USERS'
+        TOP_TEN_PROJECTS_BY_ENTRIES = _PREFIX + 'TOP-TEN-PROJECTS-BY-ENTRIES'
```
Comment (k9845) [inline deep/caches.py]: Update count or list according to return type.

**Review by k9845 — CHANGES_REQUESTED** [review]

**Comment by k9845** [comment]
@thenav56 Please also update failed test.

File: apps/deep_explore/schema.py [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+
+def count_by_type_generator(type_name):
+    return graphene.List(
+        graphene.NonNull(
+            type(type_name, (graphene.ObjectType,), {
+                'date': graphene.Date(required=True),
+                'count': graphene.Int(required=True),
+            })
+        )
+    )
```
Comment (k9845) [inline apps/deep_explore/schema.py]: This isn't used anywhere. Also, we have defined a type above.

**Review by k9845 — CHANGES_REQUESTED** [review]

File: apps/deep_explore/tasks.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+              "{_tb(Entry)}".project_id
+            ORDER BY date, project_id
+        )
+        ON CONFLICT (project_id, geo_area_id, date)
+        DO UPDATE SET
+            entries_count = EXCLUDED.entries_count;
+    """
+
+
+def update_deep_explore_entries_count_by_geo_aggreagate():
```
Comment (thenav56) [inline apps/deep_explore/tasks.py]: Add an option to clear and calculate from scratch. Run once a month
```sql
TRUNCATE TABLE someTable RESTART IDENTITY;
```

**Pull request merged**